### PR TITLE
[test/plugin] immigrate spring scenario

### PR DESCRIPTION
**Pull request by arugal**
Please answer these questions before submitting pull request

- Why submit this pull request?
- [ ] Bug fix
- [x] New feature provided
- [ ] Improve performance

- Related issues
#3583
___
### New feature or improvement
- Describe the details and related test reports.
1. For```nameStartsWith("org.apache.skywalking.").and(not(nameStartsWith("org.apache.skywalking.apm.toolkit.")));``` reasons, I use ```test.org.apache.skywalking``` as the package name.
2. The```spring 3.2.x``` does not support jdk8, so I skipped it. https://stackoverflow.com/questions/30072135/spring-3-2-x-with-java-8

### CODE REVIEW

**Comment by wu-sheng** [comment]
@dmsolr please review.

File: test/plugin/scenarios/spring-3.1.x-scenario/src/main/resources/log4j2.xml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        <logger name="com.a.eye.skywalking.ui" level="debug" additivity="false">
+            <AppenderRef ref="Console"/>
+        </logger>
+        <!--<logger name="org.skywalking.apm.dependencies" level="off">-->
+        <!--</logger>-->
+        <Root level="debug">
+            <AppenderRef ref="Console"/>
+        </Root>
+    </Loggers>
+</Configuration>
```
Comment (dmsolr) [inline test/plugin/scenarios/spring-3.1.x-scenario/src/main/resources/log4j2.xml]: It is a bad template. :( refer to [log4j2.xml in solrj-7.x-scenario](https://github.com/apache/skywalking/blob/master/test/plugin/scenarios/solrj-7.x-scenario/src/main/resources/log4j2.xml).  Becuase we don't log the `SYSTEM_OUT`.

(Maybe you set logger.level as `ERROR`.)

Comment (arugal) [reply test/plugin/scenarios/spring-3.1.x-scenario/src/main/resources/log4j2.xml]: done

**Comment by wu-sheng** [comment]
@dmsolr If this is OK for you, please give an approve, even you don't have official approve today.

**Comment by arugal** [comment]
> LGTM. @arugal How many test cases do we have today?
>
spring: 107
spring-async: 35
jetty: 63
httpclient: 14
ehcache: 19
dubbo: 14
spring-cloud-gateway: 3
solrj: 12

total: 266

File: test/plugin/scenarios/spring-3.0.x-scenario/src/main/java/test/org/apache/skywalking/apm/testcase/spring3/CaseController.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package test.org.apache.skywalking.apm.testcase.spring3;
```
Comment (dmsolr) [inline test/plugin/scenarios/spring-3.0.x-scenario/src/main/java/test/org/apache/skywalking/apm/testcase/spring3/CaseController.java]: The package name has wrong. `org.apache.skywalking.apm.testcase.*`

And others are ok for me.

Comment (arugal) [reply test/plugin/scenarios/spring-3.0.x-scenario/src/main/java/test/org/apache/skywalking/apm/testcase/spring3/CaseController.java]: Using ```org.apache.skywalking``` as the package name,  the ```SpringComponentInstrumentation、SpringRepositoryInstrumentation、SpringServicesInstrumentation```   will not work

https://github.com/apache/skywalking/blob/24cc91c9af6373a97418a011e0d27c166b88782c/apm-sniffer/apm-agent/src/main/java/org/apache/skywalking/apm/agent/SkyWalkingAgent.java#L167-L169

Comment (arugal) [reply test/plugin/scenarios/spring-3.0.x-scenario/src/main/java/test/org/apache/skywalking/apm/testcase/spring3/CaseController.java]: [spring-annotation-plugin](https://github.com/apache/skywalking/tree/master/apm-sniffer/optional-plugins/optional-spring-plugins/spring-annotation-plugin/src/main/java/org/apache/skywalking/apm/plugin/spring/annotations)

Comment (dmsolr) [reply test/plugin/scenarios/spring-3.0.x-scenario/src/main/java/test/org/apache/skywalking/apm/testcase/spring3/CaseController.java]: i got it. Thanks for your explanation.

Comment (dmsolr) [reply test/plugin/scenarios/spring-3.0.x-scenario/src/main/java/test/org/apache/skywalking/apm/testcase/spring3/CaseController.java]: i got it. Thanks for your explanation.

**Comment by dmsolr** [comment]
LGTM

**Pull request merged**